src/lmic/lmic.c

```diff
@@ -726,6 +726,24 @@ scan_mac_cmds(
             oidx += 2;
             continue;
         } /* end case */
+        case MCMD_DeviceTimeAns: {
+#if defined(LMIC_ENABLE_DeviceTimeReq)
+            // The first 4 bytes contain the seconds since the GPS epoch (i.e
+            // January the 6th 1980 at 00:00:00 UTC).
+            // Note: the octet order for all multi-octet fields is little endian
+            uint32_t seconds_since_gps_epoch;
+            memcpy(&seconds_since_gps_epoch, &opts[oidx + 1], sizeof(seconds_since_gps_epoch));
```

Comment:
this assumes that the local system is little-endian. Better to do shifts and adds.